pkgs/servers/home-assistant/custom-components/huawei_solar/package.nix

```diff
@@ -0,0 +1,33 @@
+{
+  lib,
+  buildHomeAssistantComponent,
+  fetchFromGitHub,
+  huawei-solar,
+}:
+
+buildHomeAssistantComponent rec {
+  owner = "wlcrs";
+  domain = "huawei_solar";
+  version = "1.4.1";
+
+  src = fetchFromGitHub {
+    inherit owner;
+    repo = "huawei_solar";
+    rev = version;
+    hash = "sha256-NQusnrXZOmRVVcX6a/GvwOeW62MrkypTG01+y+L6Ihs=";
+  };
+
+  patchPhase = ''
+    mkdir -p custom_components/huawei_solar && \
+    find . -mindepth 1 -maxdepth 1 -not -name custom_components -exec mv -t custom_components/huawei_solar {} +'';
```

Comment:
Remove this workaround and rebase onto master, where the installPhase should be fixed.